render-ws-spark-client/src/main/java/org/janelia/render/client/spark/pipeline/AlignmentPipelineStep.java

```diff
@@ -0,0 +1,42 @@
+package org.janelia.render.client.spark.pipeline;
+
+import java.io.IOException;
+
+import org.apache.spark.api.java.JavaSparkContext;
+
+/**
+ * Anything that can be run as part of a Spark alignment pipeline.
+ *
+ * @author Eric Trautman
+ */
+public interface AlignmentPipelineStep {
+
+    /**
+     * Validates the specified pipeline parameters are sufficient for this step.
+     *
+     * @param  pipelineParameters  parameters to validate.
+     *
+     * @throws IllegalArgumentException
+     *   if any of the parameters are invalid or missing.
+     */
+    void validatePipelineParameters(final AlignmentPipelineParameters pipelineParameters)
+            throws IllegalArgumentException;
```

Comment:
There are a few points that are not clear to me when I try to imagine how this method is used:
- Can it mutate the given `AlignmentPipelineParameters` or not? Either way, this should be made explicit, e.g. by simply stating it in the documentation, or with a `@Contract(pure = true)` annotation.
- The method does not return anything, so only the the "negative" outcome leads to observable behavior. Maybe returning a `bool` would be better? But then, `false` is redundant...

Reply:
I tried to add some more-helpful javadocs.  Bottom line is no mutation and throw exception for problem.  This is a validation pattern/approach that makes sense to me and is repeated in many other contexts.

Reply:
I wasn't familiar with that pattern, so I tried to come up with a "better" solution for several minutes, but I didn't succeed. So, I feel convinced that this should stay as it is. Your javadoc does help a lot, though!